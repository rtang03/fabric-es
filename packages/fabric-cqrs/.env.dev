CHANNEL_NAME=loanapp
CA_NAME=rca-org1
CA_ENROLLMENT_ID_ADMIN=rca-org1-admin
CA_ENROLLMENT_SECRET_ADMIN=rca-org1-adminPW
ORDERER_HOSTNAME=orderer0-org0
ORDERER_TLSCA_CERT=../../deployments/dev-net/artifacts/crypto-config/Org0MSP/orderer0.org0.com/tls-msp/tlscacerts/tls-0-0-0-0-5052.pem
CONNECTION_PROFILE=connection/connection-org1.local.yaml
MSPID=Org1MSP
ORG_ADMIN_ID=admin-org1.net
ORG_ADMIN_SECRET=admin-org1.netPW
WALLET=assets/wallet

# Org2 (pbctfp) - SOLELY used for running service.multiusers.test.ts
ORG2_ORG_ADMIN_ID=admin-org2.net
ORG2_ORG_ADMIN_SECRET=admin-org2.netPW
ORG2_MSPID=Org2MSP
ORG2_CONNECTION_PROFILE=connection/connection-org2.local.yaml
ORG1_WALLET=assets/walletOrg1
ORG2_WALLET=assets/walletOrg2
ORG2_CA_NAME=rca-org1
ORG2_CA_ENROLLMENT_ID_ADMIN=rca-org2-admin
ORG2_CA_ENROLLMENT_SECRET_ADMIN=rca-org2-adminPW



