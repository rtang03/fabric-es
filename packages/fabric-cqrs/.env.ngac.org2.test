## unsure if this is required
COLLECTION=Org1PrivateDetails
PATH_TO_CHANNEL_CONFIG=../../artifacts/channel-config

### MANDATORY
#Orgnaization Admin EnrollmentID
ORG_ADMIN_ID=Admin@org2.example.com
#Organization Admin EnrollSecret
ORG_ADMIN_SECRET=peer1pw
#URL for Orgnization CA
ORG_CA_URL=https://0.0.0.0:5055
# org name defined in connnection profile; case sensitive
ORGNAME=Org2
# Path to Fabric Network
NETWORK_LOCATION=../../network/hosts
# Path to Connection Path
CONNECTION_PROFILE=connection/peer0org2.yaml
# Relative path to wallet directory
WALLET=assets/wallet
# Channel Event Hub
CHANNEL_HUB=peer0.org2.example.com
# Optional:
CHAINCODE_ID=eventstore
CHAINCODE_ID_PRIVATEDATA=privatedata
# Used in logger
APPLICATION=peer0org2app


