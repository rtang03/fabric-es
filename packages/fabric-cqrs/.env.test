#MSPID=Org1MSP
#CHANNEL_NAME=mychannel

COLLECTION=Org1PrivateDetails
PATH_TO_CHANNEL_CONFIG=../../artifacts/channel-config
CHANNEL_NAME=eventchannel

### MANDATORY
#Orgnaization Admin EnrollmentID
ORG_ADMIN_ID=Admin@org1.example.com
#Organization Admin EnrollSecret
ORG_ADMIN_SECRET=peer1pw
#URL for Orgnization CA
ORG_CA_URL=https://0.0.0.0:5054
# org name defined in connnection profile; case sensitive
ORGNAME=Org1
# Path to Fabric Network
NETWORK_LOCATION=../../network/hosts
# Path to Connection Path
CONNECTION_PROFILE=connection/peer0org1.yaml
# Relative path to wallet directory
WALLET=assets/wallet
# Channel Event Hub
CHANNEL_HUB=peer0.org1.example.com
# Optional:
CHAINCODE_ID=eventstore
CHAINCODE_ID_PRIVATEDATA=privatedata


