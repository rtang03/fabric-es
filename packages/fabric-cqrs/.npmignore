*
!dist/
!dist/**/*
!package.json
!*.md
!LICENSE
!README.md
