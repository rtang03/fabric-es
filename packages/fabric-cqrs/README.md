# fabric-cqrs

## Installation

```shell script
yarn install @fabric-es/fabric-cqrs
```
