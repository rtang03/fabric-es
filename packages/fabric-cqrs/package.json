{
  "name": "@espresso/fabric-cqrs",
  "version": "1.0.0",
  "main": "dist/index.js",
  "typings": "dist/index.d.ts",
  "scripts": {
    "build": "yarn run clean && tsc",
    "build:prod": "yarn run clean && tsc -p ./tsconfig.prod.json",
    "package": "yarn run build:cleanup && yarn run cp:package.json && yarn run cp:dist",
    "build:cleanup": "rimraf ../../.build/packages/fabric-cqrs && mkdir -p ../../.build/packages/fabric-cqrs",
    "cp:package.json": "cp ./package.json ../../.build/packages/fabric-cqrs/",
    "cp:dist": "cp -R ./dist ../../.build/packages/fabric-cqrs/",
    "clean": "rimraf dist",
    "test:account": "jest account.unit.test",
    "test:service.integration": "jest service.integration.test",
    "test:service.multiusers": "rimraf assets/walletOrg1 && rimraf assets/walletOrg2 && jest service.multiusers.test",
    "test:service.privatedata.integration": "jest service.privatedata.integration.test",
    "test:service.ngac.crud.integration": "(broke) jest service.ngac.crud.integration.test",
    "test:service.ngac.pe.int.1u": "(broke) jest one-user.service.ngac.pe.int.test",
    "test:store.integration": "jest command.integration.test",
    "test:store.privatedata.integration": "jest command.privatedata.integration.test",
    "test:peer.integration": "jest peer.integration.test",
    "test:peer.privatedata.integration": "jest peer.privatedata.integration.test",
    "test:peer.ngac.integration": "(not implemented yet) jest peer.ngac.integration.test",
    "test:all-unit-test": "jest unit.test",
    "enrollAdmin": "ts-node src/account/enrollAdmin.ts",
    "enrollCaAdmin": "ts-node src/account/enrollCaAdmin.ts",
    "enrollAdminOrg2": "(broke) ts-node src/ngac_test/enrollAdmin.ts",
    "typedoc": "typedoc --excludeNotExported --theme markdown --exclude \"**/*+(.test|constants).ts\" --out docs src/index.ts"
  },
  "engines": {
    "node": ">=8",
    "npm": ">=5"
  },
  "private": true,
  "engineStrict": true,
  "dependencies": {
    "@espresso/operator": "1.0.0",
    "dotenv": "^8.1.0",
    "fabric-client": "^1.4.4",
    "fabric-contract-api": "^1.4.2",
    "fabric-network": "^1.4.4",
    "js-yaml": "^3.13.1",
    "lodash": "^4.17.15",
    "prettyjson": "^1.2.1",
    "redux": "^4.0.4",
    "redux-observable": "^1.2.0",
    "rxjs": "^6.5.3",
    "winston": "^3.2.1"
  },
  "devDependencies": {
    "@types/bytebuffer": "^5.0.40",
    "@types/dotenv": "^6.1.1",
    "@types/jest": "^24.0.18",
    "@types/js-yaml": "^3.12.1",
    "@types/lodash": "^4.14.138",
    "@types/node": "^8.10.54",
    "jest": "^24.9.0",
    "prettier": "^1.19.1",
    "rimraf": "^3.0.0",
    "ts-jest": "^24.0.2",
    "ts-node": "^8.3.0",
    "tslint": "^5.20.0",
    "tslint-config-prettier": "^1.18.0",
    "tslint-plugin-prettier": "^2.0.1",
    "typedoc": "^0.15.7",
    "typedoc-plugin-markdown": "^2.2.16",
    "typescript": "^3.7.2"
  }
}
