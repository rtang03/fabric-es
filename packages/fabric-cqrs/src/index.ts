/**
 * @packageDocumentation
 * @hidden
 */
export {
  createPeer,
  getMockRepository,
  getPrivatedataMockRepository,
  createProjectionDb,
  createQueryDatabase
} from './peer';
export { getContract, getNetwork, submit, evaluate, evaluate$ } from './services';
export * from './types';
export {
  dispatchResult,
  generateToken,
  getAction,
  getErrorAction,
  getErrorActionHandler,
  getSuccessAction,
  getSuccessActionHandler
} from './cqrs/utils';
export { fromCommitsToGroupByEntityId, getHistory, isCommit } from './peer/utils';
