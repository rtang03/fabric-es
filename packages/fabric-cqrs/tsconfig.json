{
  "compilerOptions": {
    "outDir": "dist",
    "target": "es2017",
    "moduleResolution": "node",
    "esModuleInterop": true,
    "module": "commonjs",
    "declaration": true,
    "sourceMap": false,
    "types": ["node", "jest"],
    "lib": ["es2017", "dom", "esnext.asynciterable"],
    "rootDir": "src",
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true
  },
  "exclude": ["node_modules", "dist"],
  "typedocOptions": {
    "entryPoints": ["src/index.ts"],
    "out": "docs",
    "name": "fabric-cqrs",
    "includeVersion": true,
    "excludeExternals": true,
    "exclude": "\"**/*+(.test|constants|index).ts\"",
    "readme": "../../README.md",
    "hideGenerator": true
  }
}
