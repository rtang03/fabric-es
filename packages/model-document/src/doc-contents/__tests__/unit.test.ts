import { buildFederatedSchema } from '@apollo/federation';
import { Commit, getPrivatedataMockRepository, getReducer, PrivatedataRepository } from '@fabric-es/fabric-cqrs';
import { DataSrc } from '@fabric-es/gateway-lib';
import { ApolloServer } from 'apollo-server';
import { createTestClient } from 'apollo-server-testing';
import gql from 'graphql-tag';
import {
  CREATE_DOC_CONTENTS,
  DocContents,
  DocContentsEvents,
  docContentsReducer,
  docContentsResolvers,
  docContentsTypeDefs,
  UPDATE_DOC_CONTENTS
} from '..';

const GET_CONTENTS_BY_ID = gql`
  query GetDocContentsById($documentId: String!) {
    getDocContentsById(documentId: $documentId) {
      documentId
      content {
        ... on Data {
          body
        }
        ... on File {
          format
          link
        }
      }
    }
  }
`;

const userId = 'unitTestUser';
const mockdb: Record<string, Commit> = {};
const docContentsRepo: PrivatedataRepository = getPrivatedataMockRepository<DocContents, DocContentsEvents>(
  mockdb,
  'docContents',
  getReducer<DocContents, DocContentsEvents>(docContentsReducer)
);

let service;

beforeAll(async () => {
  service = new ApolloServer({
    schema: buildFederatedSchema([{ typeDefs: docContentsTypeDefs, resolvers: docContentsResolvers }]),
    dataSources: () => ({
      docContents: new DataSrc({ repo: docContentsRepo })
    }),
    context: () => ({ user_id: 'admin' })
  });
});

afterAll(
  async () =>
    new Promise(done =>
      setTimeout(() => {
        console.log('DocContents Unit Test - Resolver Finished');
        done();
      }, 500)
    )
);

describe('DocContents Unit Test - Resolver', () => {
  it('create docContents 0', async () =>
    createTestClient(service)
      .mutate({
        mutation: CREATE_DOC_CONTENTS,
        variables: {
          userId,
          documentId: 'D0000',
          content: {
            body: '{ "message": "Document Contents 0" }'
          }
        }
      })
      .then(({ data }) => expect(data.createDocContents.id).toEqual('D0000'))
      .catch(_ => expect(false).toBeTruthy()));

  it('create docContents 1', async () =>
    createTestClient(service)
      .mutate({
        mutation: CREATE_DOC_CONTENTS,
        variables: {
          userId,
          documentId: 'D0001',
          content: {
            format: 'PDF',
            link: 'http://fake.it/docs/unitTestDocContents-1.pdf'
          }
        }
      })
      .then(({ data }) => expect(data.createDocContents.id).toEqual('D0001'))
      .catch(_ => expect(false).toBeTruthy()));

  it('create docContents 2', async () =>
    createTestClient(service)
      .mutate({
        mutation: CREATE_DOC_CONTENTS,
        variables: {
          userId,
          documentId: 'D0002',
          content: {
            body: '{ "message": "Document Contents 2" }'
          }
        }
      })
      .then(({ data }) => expect(data.createDocContents.id).toEqual('D0002'))
      .catch(_ => expect(false).toBeTruthy()));

  // TODO: Implement lifecycle event attribute to prevent creating same entity more than once
  // NOTE: This 'create docContents' call should return normal, but querying 'L0000' should return the original result instead of the changed values
  it('create docContents 0 again', async () =>
    createTestClient(service)
      .mutate({
        mutation: CREATE_DOC_CONTENTS,
        variables: {
          userId,
          documentId: 'D0000',
          content: {
            body: '{ "message": "Document Contents 0VERWRITTEN" }'
          }
        }
      })
      .then(({ data }) => expect(data.createDocContents.id).toEqual('D0000'))
      .catch(_ => expect(false).toBeTruthy()));

  it('update docContents 1', async () =>
    createTestClient(service)
      .mutate({
        mutation: UPDATE_DOC_CONTENTS,
        variables: {
          userId,
          documentId: 'D0001',
          content: {
            format: 'JPEG',
            link: 'http://fake.it/docs/unitTestDocContents-1.jpg'
          }
        }
      })
      .then(({ data }) => expect(data.updateDocContents.id).toEqual('D0001'))
      .catch(_ => expect(false).toBeTruthy()));

  it('update docContents 2', async () =>
    createTestClient(service)
      .mutate({
        mutation: UPDATE_DOC_CONTENTS,
        variables: {
          userId,
          documentId: 'D0002',
          content: {
            body: '{ "message": "Document Contents 2 EDITED" }'
          }
        }
      })
      .then(({ data }) => expect(data.updateDocContents.id).toEqual('D0002'))
      .catch(_ => expect(false).toBeTruthy()));

  it('change content type of docContents 1', async () =>
    createTestClient(service)
      .mutate({
        mutation: UPDATE_DOC_CONTENTS,
        variables: {
          userId,
          documentId: 'D0001',
          content: {
            body: '{ "message": "Document Contents 1!!!!!" }'
          }
        }
      })
      .then(({ errors }) =>
        expect(
          errors.reduce((acc, cur) => (cur.message.includes('DOC_CONTENTS_MISMATCHED') ? cur.message : acc), '')
        ).toContain('DOC_CONTENTS_MISMATCHED')
      )
      .catch(_ => expect(false).toBeTruthy()));

  it('create docContents with empty content', async () =>
    createTestClient(service)
      .mutate({
        mutation: CREATE_DOC_CONTENTS,
        variables: {
          userId,
          documentId: 'D9999',
          content: {}
        }
      })
      .then(({ errors }) =>
        expect(
          errors.reduce((acc, cur) => (cur.message.includes('REQUIRED_DATA_MISSING') ? cur.message : acc), '')
        ).toContain('REQUIRED_DATA_MISSING')
      )
      .catch(_ => expect(false).toBeTruthy()));

  it('update docContents 2 with empty content', async () =>
    createTestClient(service)
      .mutate({
        mutation: UPDATE_DOC_CONTENTS,
        variables: {
          userId,
          documentId: 'D0002',
          content: {}
        }
      })
      .then(({ errors }) =>
        expect(
          errors.reduce((acc, cur) => (cur.message.includes('REQUIRED_DATA_MISSING') ? cur.message : acc), '')
        ).toContain('REQUIRED_DATA_MISSING')
      )
      .catch(_ => expect(false).toBeTruthy()));

  it('update an non-existing docContents', async () =>
    createTestClient(service)
      .mutate({
        mutation: UPDATE_DOC_CONTENTS,
        variables: {
          userId,
          documentId: 'D9999',
          content: { body: 'Hello' }
        }
      })
      .then(({ errors }) =>
        expect(
          errors.reduce((acc, cur) => (cur.message.includes('DOC_CONTENTS_NOT_FOUND') ? cur.message : acc), '')
        ).toContain('DOC_CONTENTS_NOT_FOUND')
      )
      .catch(_ => expect(false).toBeTruthy()));

  it('query docContents 0 by id', async () =>
    createTestClient(service)
      .query({
        query: GET_CONTENTS_BY_ID,
        variables: { documentId: 'D0000' }
      })
      .then(({ data }) => expect(data.getDocContentsById).toMatchSnapshot())
      .catch(_ => expect(false).toBeTruthy()));

  it('query docContents 1 by id', async () =>
    createTestClient(service)
      .query({
        query: GET_CONTENTS_BY_ID,
        variables: { documentId: 'D0001' }
      })
      .then(({ data }) => expect(data.getDocContentsById).toMatchSnapshot())
      .catch(_ => expect(false).toBeTruthy()));

  it('query docContents 2 by id', async () =>
    createTestClient(service)
      .query({
        query: GET_CONTENTS_BY_ID,
        variables: { documentId: 'D0002' }
      })
      .then(({ data }) => expect(data.getDocContentsById).toMatchSnapshot())
      .catch(_ => expect(false).toBeTruthy()));
});
