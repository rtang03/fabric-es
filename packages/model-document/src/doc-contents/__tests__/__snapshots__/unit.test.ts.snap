// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocContents Unit Test - Resolver query docContents 0 by id 1`] = `
Object {
  "content": Object {
    "body": "{ \\"message\\": \\"Document Contents 0VERWRITTEN\\" }",
  },
  "documentId": "D0000",
}
`;

exports[`DocContents Unit Test - Resolver query docContents 1 by id 1`] = `
Object {
  "content": Object {
    "format": "JPEG",
    "link": "http://fake.it/docs/unitTestDocContents-1.jpg",
  },
  "documentId": "D0001",
}
`;

exports[`DocContents Unit Test - Resolver query docContents 2 by id 1`] = `
Object {
  "content": Object {
    "body": "{ \\"message\\": \\"Document Contents 2 EDITED\\" }",
  },
  "documentId": "D0002",
}
`;
